### DISCUSSION TITLE
Adding an optimized proxy for Minecraft - Velocity.

### DISCUSSION THREAD

**Discussion by d3adwolf** [GameServerManagers/LinuxGSM#4040] [Ideas]
Greetings again, I would like to offer to add an optimized proxy server core for Minecraft - Velocity. It is made by the same developers as Paper with Waterfall, so there shouldn't be any problems with communication and proxy integration in your server library.

As written on the Velocity website (https://velocitypowered.com/) - proxy kernel aims to be more productive and more flexible than Waterfall, I have not got time to test Velocity on LinuxGSM yet, by changing configs, but I will do it in the coming days and will add this "article".

For now, I ask you to support the idea with an arrow up to raise it to the top and write a comment about your experience with this proxy core. And thanks for reading! 

Also, if you are interested in an optimized kernel for a normal server, I recommend reading the idea (#4039) about Pufferfish.

**Comment by h3o66** [comment]
the velocity proxy was added with v22.2.0 and PR #3904
